### DISCUSSION TITLE
Menus reogranization options

### DISCUSSION THREAD

**Discussion by portnov** [nortikin/sverchok#4831] [General]
For a while, there have been discussions about how to make Sverchok menu categories "the right way".
Nowadays, it is quite easy for almost anyone to reconfigure the menu as you wish. You just have to edit `index.yaml` file in sverchok directory.

So, anyone can make The Right Menu of your own, try to work with it, enhance it, then share your `index.yaml` on gist.github.com and post a link here, and argue why this is the best choice. 

Will we select one of these options as default? Or will we add a parameter to sverchok settings to select between several options? Or...? We do not know yet. Let's just look what our options are first. Then we will think what to do with them.

**Comment by portnov** [comment]
For starters, here is proposal number 0: https://gist.github.com/portnov/79518dd2b679b91df7988cb6863c12e1

![Screenshot_20221220_192415](https://user-images.githubusercontent.com/284644/208691533-3e56be77-0cf2-4ba3-979b-7ff0d1469ce9.png)
![Screenshot_20221220_192341](https://user-images.githubusercontent.com/284644/208691538-21fed7ef-3282-4604-8f03-d2ee38e8dd5c.png)
![Screenshot_20221220_192230](https://user-images.githubusercontent.com/284644/208691540-35d017e6-80c6-4e71-b5df-5e1397cb72fa.png)

**Comment by nortikin** [comment]
One more thing i have acentuate:
At beginning i made naming as in viewers/lists etc nodes as next:
```C#
CATEGORY          NODE     MK2
-------------     ------- -----
|                 |       |
List               name    version
```
So, first is category name (maybe shorten) that user's eye will ignore when read names, it makes selection faster and UI cleaner.
Lets go refactoring!!!

**Reply by nortikin** [reply]
![image](https://user-images.githubusercontent.com/5783432/208769960-5294b3ca-e6df-4716-8a50-44666357fa01.png)
For example. It will be more intuitive if Curve step before names always, not in part of nodes

**Reply by nortikin** [reply]
And yes, backword compability will gone.

**Reply by rendetto** [reply]
> For example. It will be more intuitive if Curve step before names always, not in part of nodes

@nortikin Do you mean that "Intersect NURBS Curves" should be renamed to "Intersect" or "Intersect NURBS"?

**Reply by nortikin** [reply]
@rendetto no, but `Curve Intersect NURBS`

**Comment by rendetto** [comment]
> For starters, here is proposal number 0...

@portnov I'd say I'm fine with your structure. It have it's logic. The downside of it is that NURBS tools are a bit too deep, but that's me. Fortunately it's very easy to adapt the index.yaml to any work style.

I feel that reorganizing is a hard theme to do and easily could open Pandora's box. I was trying to rearrange the tools myself and found contradictions and ambiguity at fixing where certain tools should reside. For example you define "Curve Segment" as an Analyzer, where I would define it as a Modifier. 

I made this "proposal number 1": https://gist.github.com/rendetto/ea93663d9cb41c61a408098abb86ed22
based on "proposal number 0" but addressing the problem with NURBS tools being too deep by leveling them up.
![shallow_curve](https://user-images.githubusercontent.com/66558924/208781364-fd3d40f3-2018-44b3-8e75-ddb8362c0fd3.jpg)

![shallow_surface](https://user-images.githubusercontent.com/66558924/208781384-77344108-73b7-4a26-b9af-95683b939526.jpg)

**Reply by nortikin** [reply]
> > For starters, here is proposal number 0...


Than in that case we have to remove viz menu and scene, to make it lay on specific type menu
Also, my list proposal can fit your concept

**Comment by nortikin** [comment]
https://gist.github.com/nortikin/f3249b7de9c9c78b035f946cdf74ae84
My version
![image](https://user-images.githubusercontent.com/5783432/209429206-d208e44d-cbd5-4f53-8840-88b7b0e4971b.png)
![image](https://user-images.githubusercontent.com/5783432/209429259-ab0d1aa6-8201-4c83-bd7f-c5f5dbd65cd5.png)
![image](https://user-images.githubusercontent.com/5783432/209429280-90d3cf52-5b2f-44ec-ae1d-f665c370d60d.png)
![image](https://user-images.githubusercontent.com/5783432/209429303-77c9461d-a697-42db-b1c7-f48c7e6a9618.png)

**Reply by nortikin** [reply]
couple movements also, script menu moved under math

**Comment by nortikin** [comment]
And my proposal to make settings/properties to choose menue from yamls.
To check if all names are in, we have to make test for yamls, comparing all the nodes class names

**Comment by nortikin** [comment]
One more of options - to have extra yamls inside datafiles/sverchok/menues and check them also with test.

**Comment by portnov** [comment]
PR to add an option to select one of menu presets: #4841.

**Reply by portnov** [reply]
#4841 is now merged. See solution description there.
